quodlibet/util/collection.py

```diff
@@ -47,19 +48,73 @@
 ELPOEP = list(reversed(PEOPLE))
 PEOPLE_SCORE = [100 ** i for i in range(len(PEOPLE))]
 
+SMOOTHING_FACTOR_DEFAULT = 10
 
-def avg(nums):
+
+def unweighted_average(nums):
     """Returns the average (arithmetic mean) of a list of numbers"""
     return float(sum(nums)) / len(nums)
 
 
-def bayesian_average(nums, c=None, m=None):
+def unweighted_bayesian_average(nums, c=None, m=None):
     """Returns the Bayesian average of an iterable of numbers,
     with parameters defaulting to config specific to ~#rating."""
     m = m or config.RATINGS.default
     c = c or config.getfloat("settings", "bayesian_rating_factor", 0.0)
-    ret = float(m * c + sum(nums)) / (c + len(nums))
-    return ret
+    return float(m * c + sum(nums)) / (c + len(nums))
+
+
+def hyperbolic_smoother(x, n=SMOOTHING_FACTOR_DEFAULT):
+    """n determines how strongly smaller values will be magnified"""
+    return (n + 1) * x / (n * x + 1)
+
+
+def smoothed_length_weights_and_nums(songs, key, total_length, smoothing_factor):
+    """Returns smoothed length-based weights and the songs' associated numeric values"""
```

Comment:
```suggestion
    """:returns: smoothed length-based weights and the songs' associated numeric values"""
```
Though not sure we use RST docstrings much in this codebase actually (probably should)

Reply:
Happy to add them for a few functions! :)

I wonder about something. Given that our current style for multi line comments is to start on the same line, is it okay to break with this style when using RST. 

This seems far less readable:

```python
    """:param x: a float in the range [0, 1]
    :param n: determines how strongly smaller values will be magnified
    :returns: a float in the range [0, 1] with smaller values magnified
    """
```

than this:

```python
    """
    :param x: a float in the range [0, 1]
    :param n: determines how strongly smaller values will be magnified
    :returns: a float in the range [0, 1] with smaller values magnified
    """
```

Reply:
Cool, and I agree second looks better. None of our linters care enough (currently) about these things so whatever looks good I guess